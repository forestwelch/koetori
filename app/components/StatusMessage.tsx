interface StatusMessageProps {
  isRecording: boolean;
  isProcessing: boolean;
  recordingTime?: number;
}

export function StatusMessage({
  isRecording,
  isProcessing,
  recordingTime = 0,
}: StatusMessageProps) {
  const formatTime = (seconds: number) => {
    const mins = Math.floor(seconds / 60);
    const secs = seconds % 60;
    return `${mins}:${secs.toString().padStart(2, "0")}`;
  };

  const getStatus = () => {
    if (isProcessing)
      return {
        title: "Processing...",
        subtitle: "Please wait while we transcribe your recording",
      };
    if (isRecording)
      return {
        title: `Recording... ${formatTime(recordingTime)}`,
        subtitle: "Click the button to stop recording",
      };
    return {
      title: "Ready to Record",
      subtitle: "Click the microphone to start capturing audio",
    };
  };

  const status = getStatus();

  return (
    <div className="space-y-2">
      <p className="text-[#e2e8f0] text-lg font-light">{status.title}</p>
      <p className="text-[#64748b] text-sm font-light">{status.subtitle}</p>
    </div>
  );
}
