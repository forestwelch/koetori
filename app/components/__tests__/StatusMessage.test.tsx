import { render } from "@testing-library/react";
import { StatusMessage } from "../StatusMessage";

describe("StatusMessage Component", () => {
  it("renders idle state correctly", () => {
    const { container } = render(
      <StatusMessage
        isRecording={false}
        isProcessing={false}
        recordingTime={0}
      />
    );

    // Snapshot test - captures the entire rendered output
    // If anything changes, Jest will show you the diff
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders recording state correctly", () => {
    const { container } = render(
      <StatusMessage
        isRecording={true}
        isProcessing={false}
        recordingTime={15}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders processing state correctly", () => {
    const { container } = render(
      <StatusMessage
        isRecording={false}
        isProcessing={true}
        recordingTime={0}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  // Behavior test - checks that the component works correctly
  it("formats recording time correctly", () => {
    const { getByText } = render(
      <StatusMessage
        isRecording={true}
        isProcessing={false}
        recordingTime={75} // 1 minute 15 seconds
      />
    );

    // Check that "1:15" appears in the document
    expect(getByText(/1:15/)).toBeInTheDocument();
  });
});
