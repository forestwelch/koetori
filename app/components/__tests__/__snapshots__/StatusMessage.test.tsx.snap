// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StatusMessage Component renders idle state correctly 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="text-center text-sm sm:text-base text-[#94a3b8] font-light"
  role="status"
>
  Click to start recording
</div>
`;

exports[`StatusMessage Component renders processing state correctly 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="text-center text-sm sm:text-base text-[#94a3b8] font-light"
  role="status"
>
  Processing audio...
</div>
`;

exports[`StatusMessage Component renders recording state correctly 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="text-center text-sm sm:text-base text-[#94a3b8] font-light"
  role="status"
>
  <span>
    Recording...
     
    <span
      class="text-indigo-400"
    >
      0:15
    </span>
  </span>
</div>
`;
